docs/userguides/config.md

```diff
@@ -242,3 +243,35 @@ test:
   mnemonic: test test test test test test test test test test test junk
   number_of_accounts: 5
 ```
+
+## Plugin Settings
+
+To configure a plugin, use the name of the plugin followed by any of the plugin's settings.
+For example, to configure the `ape-solidity` plugin, you would do:
+
+```yaml
+solidity:
+  evm_version: paris  # Or any other setting defined in `ape-solidity`.
+```
+
+## Non-plugin settings
+
+Projects can use their own settings.
+Meaning, you can put whatever data you want in an `ape-config.yaml` file and read it in Ape.
+
+```{note}
+These types of settings lack sophisticated Pydantic validation and are limited in that respect.
+```
+
+```yaml
+my_project_key:
+  my_project_setting: "123"
+```
+
+Then, to access it (or any setting for that matter):
+
+```python
+from ape import project
+
+setting = project.config.my_project_key.my_project_setting  # "123"
```

Comment:
Maybe might want to show how you have to do post-validation?
```suggestion
setting = int(project.config.my_project_key.my_project_setting)  # 123
```

Reply:
it is a string though and I was trying to show that. I can use a different value though

Reply:
i changed it to `"my_value"` in both spots, maybe clearer.

Reply:
Question: if you put an integer, will it parse as an integer?

Reply:
In a config file, no, there is no way to know. That is what I meant in the my note I put in the docs that no validation will happen on project-level config. However if using `temp_config(value=123)`, it will be whatever type you are using, so you can do types programmatically.

Reply:
I could a weird typing mechanism to the yaml like:

```yaml
my_weird_thing:
  foo: 123  # int
```

but tbh seems overkill, but willing to do it if you think it's worth it.

Reply:
I could also just force types, like if `isnumeric()`, turn to int or if it is a bool-string name turn to a bool. I could be down with that.

Reply:
oh wait i was mistaken, it seems simple validation _does_ already happen